tests/state/test_state_api.py

```diff
@@ -63,3 +64,15 @@ def test_get_states_by_name(self, n, stub_broker, state_middleware, api_client):
         # check integrity
         for state in states:
             assert state in random_states
+
+    def test_request_cancel(self, stub_broker, stub_worker, api_client, state_middleware, cancel_backend):
+        stub_worker.pause()
+        message_id = "3141516"
```

Comment:
no need for all this. just test that cancel_backend.is_canceled(message_id) = True
and `assert res.json["result"] is True`. No need for workers and such.

Also add a test when cancel_middleware is not present, to see we get a nice error message.